lti_consumer/api.py

```diff
@@ -53,36 +53,51 @@ def get_lti_consumer(config_id=None, block=None):
         # Since the block was passed, preload it to avoid
         # having to instance the modulestore and fetch it again.
         lti_config.block = block
+    else:
+        raise Exception('Either a config_id or block is required to get or create an LTI Configuration.')
 
+    return lti_config
+
+
+def get_lti_consumer(config_id=None, block=None):
+    """
+    Retrieves an LTI Consumer instance for a given configuration.
+
+    Returns an instance of LtiConsumer1p1 or LtiConsumer1p3 depending
+    on the configuration.
+    """
     # Return an instance of LTI 1.1 or 1.3 consumer, depending
     # on the configuration stored in the model.
-    return lti_config.get_lti_consumer()
+    return _get_lti_config(config_id, block).get_lti_consumer()
 
 
 def get_lti_1p3_launch_info(config_id=None, block=None):
     """
     Retrieves the Client ID, Keyset URL and other urls used to configure a LTI tool.
     """
-    if not config_id and not block:
-        raise Exception("You need to pass either a config_id or a block.")
+    # Retrieve LTI Config
+    lti_config = _get_lti_config(config_id, block)
```

Comment:
@giovannicimolin, on previous implementation we could get launch info by passing either a ``config_id`` or a ``block``. But new implementation uses ``block`` just for [caching purposes](https://github.com/edx/xblock-lti-consumer/pull/125/files#diff-6e408862179502a0ac809fab914e028b18311a17a3b0874469bd6416bc4a449fR55). This changes old behavior. Is/will there be any case that we might only have ``block`` but not ``config_id`` to get launch info?

Reply:
@shimulch The behavior is exactly the same as previously, and the only thing that changed is that I have pushed the following code into a reusable function (that will be used extensively).

I've moved this code to `get_lti_config`:
```
    if config_id:
        lti_config = LtiConfiguration.objects.get(pk=config_id)
    elif block:
        lti_config = _get_or_create_local_lti_config(
            block.lti_version,
            block.location,
        )
        # Since the block was passed, preload it to avoid
        # having to instance the modulestore and fetch it again.
        lti_config.block = block
```

Reply:
@shimulch Missed your last question:

>Is/will there be any case that we might only have block but not config_id to get launch info?

Every time the API is called from whiting the XBlock, it will **only** pass the `block`, that's because there's no LTI config id storage in the LTI XBlock for now. LTI APIs only use LTI Config ID in the other hand.

We want to get rid of the Block requirement for XBlock launches, but that's on a future implementation.